flink-kubernetes/src/main/java/org/apache/flink/kubernetes/highavailability/KubernetesMultipleComponentLeaderElectionHaServices.java

```diff
@@ -69,75 +59,86 @@ public class KubernetesMultipleComponentLeaderElectionHaServices extends Abstrac
 
     private final String lockIdentity;
 
-    private final FatalErrorHandler fatalErrorHandler;
-
-    @Nullable
-    @GuardedBy("lock")
-    private DefaultMultipleComponentLeaderElectionService multipleComponentLeaderElectionService =
-            null;
-
     KubernetesMultipleComponentLeaderElectionHaServices(
             FlinkKubeClient kubeClient,
-            Executor executor,
-            Configuration config,
-            BlobStoreService blobStoreService,
-            FatalErrorHandler fatalErrorHandler)
-            throws IOException {
+            Executor ioExecutor,
+            Configuration configuration,
+            BlobStoreService blobStoreService)
+            throws Exception {
+        this(
+                kubeClient,
+                ioExecutor,
+                configuration.get(KubernetesConfigOptions.CLUSTER_ID),
+                configuration,
+                blobStoreService);
+    }
+
+    private KubernetesMultipleComponentLeaderElectionHaServices(
+            FlinkKubeClient kubeClient,
+            Executor ioExecutor,
+            String clusterId,
+            Configuration configuration,
+            BlobStoreService blobStoreService)
+            throws Exception {
+        this(
+                kubeClient,
+                kubeClient.createConfigMapSharedWatcher(
+                        KubernetesUtils.getConfigMapLabels(
+                                clusterId, LABEL_CONFIGMAP_TYPE_HIGH_AVAILABILITY)),
+                Executors.newCachedThreadPool(
+                        new ExecutorThreadFactory("config-map-watch-handler")),
+                ioExecutor,
+                clusterId,
+                UUID.randomUUID().toString(),
+                configuration,
+                blobStoreService);
+    }
 
+    private KubernetesMultipleComponentLeaderElectionHaServices(
```

Comment:
not sure why we need 2 private constructors.

Reply:
Fair enough. I'm gonna remove the intermediate one. Initially, I created it because I wanted to avoid having `configuration.get(KubernetesConfigOptions.CLUSTER_ID)` being called twice for the `clusterId`.